legacy/builder/setup_build_properties.go

```diff
@@ -126,6 +127,14 @@ func (s *SetupBuildProperties) Run(ctx *types.Context) error {
 
 	buildProperties.Merge(ctx.PackageManager.CustomGlobalProperties)
 
+	keychainProp := buildProperties.ContainsKey("build.keys.keychain")
+	signProp := buildProperties.ContainsKey("build.keys.sign_key")
+	encryptProp := buildProperties.ContainsKey("build.keys.encrypt_key")
+	// we verify that all the properties for the secure boot keys are defined or none of them is defined.
+	if !(keychainProp || signProp || encryptProp) && (keychainProp && signProp && encryptProp) {
```

Comment:
```suggestion
	if (keychainProp || signProp || encryptProp) && !(keychainProp && signProp && encryptProp) {
```
Logic error in check made it always evaluate as false.

Reply:
You are right, I redid the table of truth and I made a mistake indeed. Thanks a lot for spotting it